File: data/java/src/function_Pydantic_basemodel.java
package com.example;

import org.json.JSONArray;
import org.json.JSONObject;
import java.util.regex.Pattern;

public class UserValidator {
    public String create_user(String name, int age, String email) {
        class User {
            private final String name;
            private final int age;
            private final String email;

            public User(String name, int age, String email) {
                this.name = name;
                this.age = age;
                this.email = email;
            }

            public JSONArray validate() {
                JSONArray errors = new JSONArray();

                if (name == null || name.trim().isEmpty()) {
                    errors.put(createError("name", "name cannot be empty", "value_error"));
                }

                if (age < 0) {
                    errors.put(createError("age", "age must be positive", "value_error"));
                }

                if (email == null || email.trim().isEmpty()) {
                    errors.put(createError("email", "email cannot be empty", "value_error"));
                } else if (!isValidEmail(email)) {
                    errors.put(createError("email", "invalid email format", "value_error"));
                }

                return errors;
            }

            private boolean isValidEmail(String email) {
                String emailRegex = "^[A-Za-z0-9+_.-]+@(.+)$";
                Pattern pattern = Pattern.compile(emailRegex);
                return pattern.matcher(email).matches();
            }

            private JSONObject createError(String field, String message, String type) {
                JSONObject error = new JSONObject();
                JSONArray loc = new JSONArray();
                loc.put(field);
                error.put("loc", loc);
                error.put("msg", message);
                error.put("type", type);
                return error;
            }

            @Override
            public String toString() {
                return String.format("name='%s' age=%d email='%s'", name, age, email);
            }
        }

        try {
            User user = new User(name, age, email);
            JSONArray validationErrors = user.validate();

            if (validationErrors.length() > 0) {
                return validationErrors.toString();
            }

            return user.toString();
        } catch (Exception e) {
            JSONArray errors = new JSONArray();
            errors.put(new JSONObject()
                    .put("loc", new JSONArray().put("unknown"))
                    .put("msg", e.getMessage())
                    .put("type", "value_error"));
            return errors.toString();
        }
    }
}

File: code/generate_strategies/IR(pseudocode)/cpp/cpp/function_Pydantic_basemodel.cpp
#include <string>
#include <nlohmann/json.hpp>
#include <stdexcept>
#include <regex>

class UserValidator {
public:
    std::string create_user(const std::string& name, int age, const std::string& email) {
        nlohmann::json errors = nlohmann::json::array();

        
        if (age <= 0) {
            nlohmann::json error;
            error["loc"] = {"age"};
            error["msg"] = "value is not a valid integer";
            error["type"] = "type_error.integer";
            errors.push_back(error);
        }

        
        std::regex email_regex(R"((\w+)(\.{1}\w+)*@(\w+)(\.\w+)+)");
        if (!std::regex_match(email, email_regex)) {
            nlohmann::json error;
            error["loc"] = {"email"};
            error["msg"] = "value is not a valid email address";
            error["type"] = "type_error.email";
            errors.push_back(error);
        }

        
        if (name.empty()) {
            nlohmann::json error;
            error["loc"] = {"name"};
            error["msg"] = "value is not a valid string";
            error["type"] = "type_error.string";
            errors.push_back(error);
        }

        if (!errors.empty()) {
            return errors.dump();
        }

        std::ostringstream result;
        result << "name='" << name << "' age=" << age << " email='" << email << "'";
        return result.str();
    }
};